crates/zksync/core/src/convert/eip712.rs

```diff
@@ -133,6 +133,27 @@ impl Transaction for Eip712SignableTransaction {
     fn kind(&self) -> alloy_primitives::TxKind {
         alloy_primitives::TxKind::Call(self.inner.to.to_address())
     }
+
+    // got from EIP4844 impl
+    fn effective_gas_price(&self, base_fee: Option<u64>) -> u128 {
+        let max_fee_per_gas = self.inner.max_fee_per_gas.low_u128();
+        let max_priority_fee_per_gas = self.inner.max_priority_fee_per_gas.low_u128();
+        base_fee.map_or(max_fee_per_gas, |base_fee| {
+            // if the tip is greater than the max priority fee per gas, set it to the max
+            // priority fee per gas + base fee
+            let tip = max_fee_per_gas.saturating_sub(base_fee as u128);
+            if tip > max_priority_fee_per_gas {
+                max_priority_fee_per_gas + base_fee as u128
+            } else {
+                // otherwise return the max fee per gas
+                self.inner.max_fee_per_gas.low_u128()
+            }
+        })
+    }
+
+    fn is_dynamic_fee(&self) -> bool {
+        true
+    }
```

Comment:
Would be deleted code when you merge `main`

Reply:
I know but still need it to make this compile for now